core/src/js/components/main-nav/Nav.js

```diff
@@ -47,6 +48,10 @@ export default class Nav {
       }
     );
 
+    // add custom events to alert others when the mobile nav opens or closes
```

Comment:
```suggestion
    // Add custom events to alert others when the mobile nav opens or closes.
```

Reply:
Can we move this new block up to line 35/36 so that the instance property assignments are all together before the logic statements?

Reply:
The only logic that's above it is initializing `this.items`. I like to keep the initializing of properties next to the declarations. That way I only have to look in 1 place when I need to know what the property is.

Reply:
In javascript variables are hoisted. See: https://www.w3schools.com/js/js_hoisting.asp

It is best practices to always declare your variables and properties at the top. 

> To avoid bugs, always declare all variables at the beginning of every scope.
> Since this is how JavaScript interprets the code, it is always a good rule.

```
    this.elem = elem;
    this.topNav = this.getTopNav();
    
    // Add custom events to alert others when the mobile nav opens or closes
    this.openEvent = createEvent('openNav'); // dispatched in this.openMobileNav()
    this.closeEvent = createEvent('closeNav'); // dispatched in this.closeMobileNav()

    this.toggle = elem.querySelector(elem.tagName + ' > button');
    this.toggleText = this.toggle ? this.toggle.innerText : '';
    this.items = [];

    // If this is a subnav, we need the correpsonding HTMLElement for
    // .querySelector()
    if (elem instanceof NavItem) {
      elem = elem.item;
    }
   
    let items = elem.querySelectorAll(elem.tagName + ' > ul > li');
    items.forEach(
      item => {
        this.items.push(new NavItem(item, this));
      }
    );
```

Reply:
I updated the constructor per your request.